### DISCUSSION TITLE
Will react 19 have any significant effect or implications for Jotai?

### DISCUSSION THREAD

**Discussion by Aubrey-Russell** [pmndrs/jotai#2709] [General]
What impacts will react 19 have on Jotai and its development?

**Comment by dai-shi** [comment]
Nothing significant that we are aware of.

**Reply by Aubrey-Russell** [reply]
What about new features, something like an optimistic atom or hook that's based on an API promise or something? So similar to the useOptimistic hook , you could have useOptimisticAtom that shows a different state while an async promise is underway. This could be a nice feature that's in alignment with the react 19 design and the excellent integration of Jotai atoms with suspense. The example from react I found the most convincing:

`For example, when a user types a message into the form and hits the “Send” button, the useOptimistic Hook allows the message to immediately appear in the list with a “Sending…” label, even before the message is actually sent to a server. This “optimistic” approach gives the impression of speed and responsivenes`

**Reply by dai-shi** [reply]
Can you create `useOptimisticAtom` with `useOptimistic`? Looking forward to your experiment.
